src/dex-ui/pages/dao/ProposalDetailsPage/GovernanceProposalConfirmationDetails.tsx

```diff
@@ -146,62 +150,56 @@ export function GovernanceProposalConfirmationDetails(props: GovernanceProposalC
           )}
         </>
-        {isAuthor && (
-          <AlertDialog
-            openModalComponent={
-              <Button variant="secondary" textStyle="h3" width="290px">
-                Cancel
-              </Button>
-            }
-            title="Cancel Proposal"
-            body={
-              <Flex flexDirection="column" gap="1.25rem">
-                <Text textStyle="b1" fontSize="1rem">
-                  {`You’re about to cancel ${proposal?.title}.
-                 If you do this, ${tokenSymbol} tokens will be refunded to you and to
-            anyone that voted.`}
-                </Text>
-                <Flex direction="row">
-                  <Text flex="1" textStyle="b3" color={Color.Grey_02}>
-                    {`${tokenSymbol} Tokens Refunded to You`}
-                  </Text>
-                  <Text flex="1" textStyle="b3" textAlign="right">
-                    {votingPower}
-                  </Text>
-                </Flex>
-              </Flex>
-            }
-            footer={
-              <Flex flexDirection="column" width="100%" gap="0.5rem">
-                <Button onClick={handleCancelProposalClicked}>Cancel Proposal</Button>
-                <Button
-                  variant="secondary"
-                  onClick={() => setDialogState({ ...dialogState, isCancelProposalOpen: false })}
-                >
-                  Don't Cancel
-                </Button>
-              </Flex>
-            }
-            alertDialogOpen={dialogState.isCancelProposalOpen}
-            onAlertDialogOpen={() => setDialogState({ ...dialogState, isCancelProposalOpen: true })}
-            onAlertDialogClose={() => setDialogState({ ...dialogState, isCancelProposalOpen: false })}
+        {isAuthor ? (
+          <ProposalCancelModal
+            title={proposal?.title}
+            tokenSymbol={tokenSymbol}
+            votingPower={votingPower}
+            handleCancelProposalClicked={handleCancelProposalClicked}
+            setDialogState={setDialogState}
+            dialogState={dialogState}
           />
+        ) : (
+          <></>
         )}
       </Flex>
     ),
     [ProposalStatus.Queued]: (
       <>
         {isAuthor && (
-          <Flex>
+          <>
             <Button variant="primary" onClick={() => handleClickExecuteTransaction()}>
               Execute
             </Button>
-          </Flex>
+            <ProposalCancelModal
+              title={proposal?.title}
+              tokenSymbol={tokenSymbol}
+              votingPower={votingPower}
+              handleCancelProposalClicked={handleCancelProposalClicked}
+              setDialogState={setDialogState}
+              dialogState={dialogState}
+            />
+          </>
         )}
       </>
     ),
     [ProposalStatus.Success]: <></>,
-    [ProposalStatus.Failed]: <></>,
+    [ProposalStatus.Failed]: (
+      <>
+        {isAuthor && state !== ProposalState.Canceled ? (
+          <ProposalCancelModal
```

Comment:
In the event of `Defeated` state user should be able to Cancel the Proposal.

Reply:
Updated the Text for stepper and Proposal Card for the Cancelled state